src/plugins/github/fetchGithubRepo.js

```diff
@@ -45,15 +45,28 @@ export default function fetchGithubRepo(
     throw new Error(`Invalid token: ${token}`);
   }
 
-  const query = stringify.body(createQuery(), inlineLayout());
+  const query = createQuery();
   const variables = createVariables(repoOwner, repoName);
   const payload = {query, variables};
-  return postQuery(payload, token);
+  return postQueryExhaustive(
+    (somePayload) => postQuery(somePayload, token),
+    payload
+  ).then((x) => {
+    ensureNoMorePages(x);
```

Comment:
if we were writing this from scratch today (i.e. after writing the query pagination solution), do you think we would have written an ensureNoMorePages function here? i'm happy to keep it either way (yay defensive coding), but it's maybe an interestingly path-dependent artifact

Reply:
I would certainly include it. The cost—computationally, conceptually, in terms of codebase size—is trivial, and the value is high. I would consider moving it somewhere else, though: perhaps folded in as an assertion at the end of the `postQueryExhaustive` routine.